### PR TITLE
Fixing spaces papercuts

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/element-web/issues/16777
Fixes https://github.com/vector-im/element-web/issues/16784

image
image

### CODE REVIEW

**Comment by t3chguy** [comment]
(abusing netlify builder)

File: src/components/structures/SpaceRoomDirectory.tsx [hunk 1]
```diff
@@ -550,34 +535,78 @@ const SpaceRoomDirectory: React.FC<IProps> = ({ space, initialText = "", onFinis
...
+    </>;
+};
+
+interface IProps {
+    space: Room;
+    initialText?: string;
+    onFinished(): void;
+}
+
+const SpaceRoomDirectory: React.FC<IProps> = ({ space, onFinished, initialText }) => {
```
Comment (turt2live) [inline src/components/structures/SpaceRoomDirectory.tsx:567]: it's a little awkward that the thing mentioned in the file name is 567 lines down - would be good to separate the components (in a future PR if useful)

File: src/components/structures/SpaceRoomDirectory.tsx [hunk 2]
```diff
@@ -550,34 +535,78 @@ const SpaceRoomDirectory: React.FC<IProps> = ({ space, initialText = "", onFinis
...
+    const onCreateRoomClick = () => {
+        dis.dispatch({
+            action: 'view_create_room',
+            public: true,
+        });
+        onFinished();
+    };
+
+    const title = <React.Fragment>
+        <RoomAvatar room={space} height={32} width={32} />
```
Comment (turt2live) [inline src/components/structures/SpaceRoomDirectory.tsx:577]: I so desperately want to change this to be `<RoomAvatar square={32} />`, but that's not a concern for you or this PR :p

File: src/components/views/dialogs/SpaceSettingsDialog.tsx [hunk 3]
```diff
@@ -126,8 +126,8 @@ const SpaceSettingsDialog: React.FC<IProps> = ({ matrixClient: cli, space, onFin
             <div>
                 { _t("Make this space private") }
                 <ToggleSwitch
-                    checked={joinRule === "private"}
-                    onChange={checked => setJoinRule(checked ? "private" : "invite")}
+                    checked={joinRule !== "public"}
```
Comment (turt2live) [inline src/components/views/dialogs/SpaceSettingsDialog.tsx:129]: these should be an enum eventually

File: src/components/views/spaces/SpaceTreeLevel.tsx [hunk 4]
```diff
@@ -33,22 +33,17 @@ import {toRightOf} from "../../structures/ContextMenu";
 import {
     shouldShowSpaceSettings,
     showAddExistingRooms,
-    showCreateNewRoom,
+    showCreateNewRoom, showSpaceInvite,
```
Comment (turt2live) [inline src/components/views/spaces/SpaceTreeLevel.tsx:36]: So I'm not the only one with that webstorm bug...

**Review by turt2live — APPROVED** [review]
overall looks fine - just some future concerns, and a super tiny IDEA lint fix

File: src/components/structures/SpaceRoomDirectory.tsx (hunk 1 above)
Comment (t3chguy) [reply src/components/structures/SpaceRoomDirectory.tsx:567]: So right now we're actually not using SpaceRoomDirectory at all, as a one-commit trial into using the Space Home for it. That choice may be reverted, but in an ideal world SpaceRoomDirectory dies and SpaceHierarchy lives on in its own file. Its a temporary evil to have a less conflicty possible future revert

File: src/components/views/dialogs/SpaceSettingsDialog.tsx (hunk 3 above)
Comment (t3chguy) [reply src/components/views/dialogs/SpaceSettingsDialog.tsx:129]: Yup, more js-sdk enums ftw <3

**Pull request merged**